Library/OcBootManagementLib/BootArguments.c

```diff
@@ -360,3 +374,226 @@ OcCheckArgumentFromEnv (
 
   return HasArgument;
 }
+
+EFI_STATUS
+OcParseLoadOptions (
```

Comment:
Shall we merge this with XNU load options parsing? See OcGetArgumentFromCmd and related.

Reply:
Can do... again this would add some alloc and code overhead to things which currently use OcGetArgumentFromCmd...

Reply:
Well, maybe we can refactor the existing functions, but I'd not try to overoptimise without a really big need.

Reply:
Ah. This might be a good reason why not use the new functions, though: https://github.com/acidanthera/OpenCorePkg/blob/master/Library/OcBootManagementLib/BootArguments.c#L321-L323 - new stuff is 100% destructive/assumes ownership of options area.

Maybe rename existing method to OcSafeGetArgumentsFromCmd and add comment that it is non-desructive?

Reply:
Well, my primary point is to use common parsing code. APIs can vary.

Reply:
My parsing code is destructive (shifts memory to handle `'\'` escaped characters and inserts `'\0'` for ends of names and values). I believe cannot be used as is on options which are to be passed on to someone else. However, could be used if we make of copy of the options (which it seems we do anyway, at around the point I linked to).

Reply:
Have added to to-do list in main PR post - could be moved into an issue and done after merge?